TGM/src/main/java/network/warzone/tgm/command/CycleCommands.java

```diff
@@ -504,11 +501,27 @@ public static void team(CommandContext cmd, CommandSender sender) {
                 } else {
                     sender.sendMessage(ChatColor.RED + "/team size (team) (min) (max)");
                 }
+            } else if (cmd.getString(0).equalsIgnoreCase("shuffle")) {
+                List<MatchTeam> teams = TGM.get().getModule(TeamManagerModule.class).getTeams().stream().filter(matchTeam ->
+                        !matchTeam.isSpectator()
+                ).distinct().collect(Collectors.toList());
+
+                List<Player> players = Bukkit.getOnlinePlayers().stream().filter(player ->
```

Comment:
Instead of getting every online player, you could map each team from `teams` to player lists and collect them